browser/js/common/factories/ExportFactory.js

```diff
@@ -0,0 +1,149 @@
+app.factory('ExportFactory', function(GridFactory) {
+
+  var ExportFactory = {};
+
+    // for html generator
+    var bits = {
+        htmlopen: '<html>',
+        htmlclose: '</html>',
+        headopen: '<head>',
+        headclose: '</head>',
+        bodyopen: '<body>',
+        bodyclose: '</body>',
+        bootstrapCDN: '<link rel="stylesheet" href="http://netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css">',
+        container: '<div class="container">',
+        row: '<div class="row">',
+        close: '</div>'
+    };
+
+  function colMaker(sz, span) {
+    return '<div class="col-' + sz + '-' + span + '">';
+  }
+
+  function offsetMaker(sz, span) {
+    return '<div class="col-' + sz + '-offset-' + span + '">';
+  }
+
+  // is this node a child of the same parent
+  function isChildOfSameParent(node, parentId) {
+    return (node.parentId === parentId);
+  };
+
+  // key is parent grid's Id, value is array of its child nodes
+  function makeParentObject() {
+    var visitedGrids = {};
+    GridFactory.savedGrid.forEach(function(node, index){
+      if (!visitedGrids[node.parentId]) {
+        visitedGrids[node.parentId] = [];
+        visitedGrids[node.parentId].push(node);
+      } else {
+        visitedGrids[node.parentId].push(node);
+      }
+    })
+    return markGrids(visitedGrids);
+  };
+
+  // mark children nodes that are grids as grids
+  function markGrids(visitedGrids) {
+    var gridNumArray = [];
+    for (var grid in visitedGrids) {
+      gridNumArray.push(grid.slice(4));
+    }
+    GridFactory.savedGrid.forEach(function(node){
+      if (gridNumArray.indexOf(node.id) >= 0) {
+        node.grid = true;
+      }
+    })
+    return visitedGrids;
+  }
+
+  function separateRows(nodesArr) {
+  // Modifies nodesArr to be an array of arrays where each subarray is a ROW of nodes
+    var rowsArray = [];
+    for (var i = 0; i < nodesArr.length; i++) {
+      if (rowsArray[nodesArr[i].y] == undefined) {
+        rowsArray[nodesArr[i].y] = [nodesArr[i]];
+      } else {
+        rowsArray[nodesArr[i].y].push(nodesArr[i]);
```

Comment:
Store nodes with the same y value together as a sub-array (under the key y) as they'll be in the same row.